Otus-DevOps-2020-02/DmitryKorlas_microservices
# Monitoring 1

# Completed the homework №19


## What were done:
 - describe prometheus running in README.md
 - reorganize directory structure: detach docker things into the separate 'docker' directory
 - disable build phase in docker-compose.yml due to build is detached from docker-compose
 - connect prometheus to the puma application
 - add basic steps description about how to check services status using prometheu
 - connect node_exporter
    - see machine metrics using prometheus node_exporter
    - add description into README.md how to work with node_exporter
 

## How to run the project:
 - build services
   ```shell
   for i in ui post-py comment; do cd src/$i; bash docker_build.sh; cd -; done
   cd monitoring/prometheus
   docker build -t <YOUR_DOCKERHUB_LOGIN>/prometheus .
   ```
  - start docker
    ```shell
    cd docker
    docker compose up -d
    ```
   
## How to check, that it works correctly:
 - using IP of docker host, visit the link http://DOCKER_HOST_IP:9090
 - add chart with the `ui_health` metrics to observe the service is UP
 - try to stop any app service, say *post* or *comment*: `docker-compose stop post`
 - check chart, it display that service is affected
 - start service again `docker-compose stop post`
 - check chart, the service should be UP
-----
 Results pushed to the dockerhub:
 - https://hub.docker.com/repository/docker/dmitrykorlas/ui
 - https://hub.docker.com/repository/docker/dmitrykorlas/comment
 - https://hub.docker.com/repository/docker/dmitrykorlas/post
 - https://hub.docker.com/repository/docker/dmitrykorlas/prometheus

## PR checklist:
 - [x] Added a label with the number pf homework
